app/exercises/five.py

```diff
@@ -65,41 +71,112 @@ class CustomToolbar(NavigationToolbar2Tk):
 
         self.toolbar = CustomToolbar(self.canvas, frame)
         self.toolbar.update()
-        
+
         self.canvas._tkcanvas.pack(side=tk.TOP, fill=tk.BOTH, expand=True)
         self.canvas.draw()
 
         frame.grid_columnconfigure(0, weight=1)
         frame.grid_rowconfigure(0, weight=1)
+    
+    def append_layers_info(self):
+        layers_string = '\nWarstwy:\n'
+        for layer, vertices in self.layers.items():
+            layers_string += f'{layer}: {vertices}\n'
+
+        self.append_text(layers_string)
 
     def generate_network(self):
         try:
             n = int(self.vertices_entry.get())
         except ValueError:
-            messagebox.showinfo(title='Wykrzyknik!', message='Wprowadź prawidłowe dane wejściowe!')
+            messagebox.showinfo(title='Wykrzyknik!',
+                                message='Wprowadź prawidłowe dane wejściowe!')
             return
 
         if n < 2:
-            messagebox.showinfo(title='Wykrzyknik!', message='Liczba wierzchołków nie może być mniejsza niż 2!')
+            messagebox.showinfo(
+                title='Wykrzyknik!', message='Liczba wierzchołków nie może być mniejsza niż 2!')
             return
 
         self.graph = WeightedDirectedAdjacencyList.init_empty()
-        layers = self.graph.generate_flow_network(n)
+        self.layers = self.graph.generate_flow_network(n)
 
         self.visualization = self.graph.convert_to_networkX()
         self.axis.clear()
-        
-        labels = nx.get_edge_attributes(self.visualization,'weight')
-        pos=nx.spring_layout(self.visualization)
-        nx.draw_networkx(self.visualization, pos=pos, ax=self.axis)
-        nx.draw_networkx_edge_labels(self.visualization, pos=pos, edge_labels=labels, ax=self.axis)
+
+        labels = nx.get_edge_attributes(self.visualization, 'weight')
+        self.pos = nx.spring_layout(self.visualization)
+        nx.draw_networkx(self.visualization, pos=self.pos, ax=self.axis)
+        nx.draw_networkx_edge_labels(
+            self.visualization, pos=self.pos, edge_labels=labels, ax=self.axis)
 
         self.canvas.draw()
         self.print_graph()
+        self.append_layers_info()
+
+    def BFS(self, s, t, path, matrix):
```

Comment:
I think "source" and "target" can be better than "s" and "t"

Reply:
lower case method names please